### PR TITLE
GH-104405: Add missing PEP 523 checks

### PR DESCRIPTION
**Pull request by brandtbucher**
Also, increase our test coverage of these instructions (and add a helpful assert for future ones).

This will need to be manually backported to 3.11.

* Issue: gh-104405

### CODE REVIEW

File: Lib/test/test_capi/test_misc.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import _thread
 import importlib.machinery
 import importlib.util
+import itertools
```
Comment (AlexWaygood) [inline Lib/test/test_capi/test_misc.py]: Unused import? :)

```suggestion
```

Comment (brandtbucher) [reply Lib/test/test_capi/test_misc.py]: Indeed!

**Review by markshannon — APPROVED** [review]
I assume the new tests fail on main.

**Comment by brandtbucher** [comment]
> I assume the new tests fail on main.

Yep.

Perf came back [about 1% faster](https://github.com/faster-cpython/benchmarking-public/blob/main/results/bm-20230511-3.12.0a7%2B-26d9521/bm-20230511-linux-x86_64-brandtbucher-eval_frame-3.12.0a7%2B-26d9521-vs-base.png), thanks to our friend `gc_traversal`... 🙄

**Pull request merged**